main/src/ca/uwaterloo/flix/api/lsp/LspServer.scala

```diff
@@ -98,16 +108,89 @@ object LspServer {
       *
       * During the initialization, we should:
       * - Store the client capabilities.
+      * - Load all Flix resources, including source files JAR files, flix.toml and flix package files.
       * - Return the server capabilities.
       */
     override def initialize(initializeParams: InitializeParams): CompletableFuture[InitializeResult] = {
       System.err.println(s"initialize: $initializeParams")
 
       clientCapabilities = initializeParams.getCapabilities
 
+      loadFlixProject(initializeParams.getWorkspaceFolders)
+
       CompletableFuture.completedFuture(new InitializeResult(mkServerCapabilities()))
     }
 
+    private def getRecursiveFilesIterator(path: Path): Iterator[Path] = {
+      if (Files.exists(path) && Files.isDirectory(path))
+        Files.walk(path).iterator().asScala
+      else
+        Iterator.empty
+    }
+
+    private def checkExt(p: Path, expectedExt: String): Boolean = {
+      Files.isRegularFile(p) && p.getFileName.toString.endsWith(expectedExt)
+    }
+    /**
+      * Loads all Flix resources in the workspace, including:
+      *   - Flix source files (*.flix, src/**/*.flix, test/**/*.flix).
+      *   - JAR files (lib/**/*.jar).
+      *   - Flix package files (lib/**/*.fpkg).
+      *   - flix.toml file (from the root of the workspace).
+      */
+    private def loadFlixProject(roots: util.List[WorkspaceFolder]): Unit = {
```

Comment:
I think this function can be further improved by breaking it into smaller parts.